src/som/vmobjects/SSymbol.java

```diff
@@ -40,9 +41,13 @@
   public SSymbol(final String value) {
     string = value;
     numberOfSignatureArguments = determineNumberOfSignatureArguments();
-    if (VmSettings.KOMPOS_TRACING) {
+    if (VmSettings.KOMPOS_TRACING || VmSettings.ACTOR_TRACING
+        || VmSettings.TRACK_SNAPSHOT_ENTITIES) {
       symbolId = (short) idGenerator.getAndIncrement();
       TracingBackend.logSymbol(this);
+      if (VmSettings.TRACK_SNAPSHOT_ENTITIES) {
+        SnapshotBackend.registerSymbol(this);
```

Comment:
Hmm, I am not a huge fan that we have now two things doing the same thing roughly.

Can this be avoided?